### ISSUE TITLE
Masked_regions_output + metadata not saving properly.

### ISSUE THREAD

**Issue by rjp5nc** [awedwards/daphnia#39]
In doing the most recent test, we were going to do the first 20 or so of xaa and xfp to see if it gave any errors, and it didn't. However, it appears that the metadata output file isn't saving the blur, and the masked_regions_output only saved one of the maskings that I did, and nothing else. I believe I saved right after it, so it may be a case of not adding on the additional maskings created after the original save. The outputs should all be marked xaaRPreduced or xfpRPreduced in /mnt/spicy_4/daphnia/analysis/analysis_results. Dörthe is testing this again right now, and Ill also double check it.

**Comment by rjp5nc** [comment]
Also, I just realized from double checking, but the corresponding shape output for xaa does indeed have several images that go over 700 (i.e. full_100124_D10_50_1.0_2A_RigA_20170527T181949.bmp and a few others)

**Comment by beckerdoerthe** [comment]
Ok, Austin, that's what we discussed at some point. There are a number of pics that - as far as I recall - you were not able to fit a shape w/ 700 checkpoints. Do you think that might be causing some of the issues?  As far as I can tell, this is the list of pics with >700 points

                                                                            filepath
 1:     /mnt/spicy_4/daphnia/data/full_100034_D10_81_0.1_1A_RigA_20170527T174418.bmp
 2:    /mnt/spicy_4/daphnia/data/full_100077_D10_79_0.25_1B_RigA_20170527T180431.bmp
 3:     /mnt/spicy_4/daphnia/data/full_100124_D10_50_1.0_2A_RigA_20170527T181949.bmp
 4:   /mnt/spicy_4/daphnia/data/full_100325_AD8_30_juju4_1D_RigB_20170603T120023.bmp
 5:  /mnt/spicy_4/daphnia/data/full_100923_D10_A13_juju2_1C_RigB_20170615T115845.bmp
 6:    /mnt/spicy_4/daphnia/data/full_110265_D8_213_ctrl_1B_RigA_20170815T164029.bmp
 7:    /mnt/spicy_4/daphnia/data/full_110601_D8_183_ctrl_1B_RigB_20171005T114231.bmp
 8:    /mnt/spicy_4/daphnia/data/full_110603_D8_183_ctrl_2B_RigB_20171005T114804.bmp
 9:    /mnt/spicy_4/daphnia/data/full_110613_D8_191_juju_1B_RigB_20171004T104057.bmp
10:    /mnt/spicy_4/daphnia/data/full_110615_D8_191_juju_2B_RigB_20171004T104642.bmp
11:    /mnt/spicy_4/daphnia/data/full_110647_D8_327_juju_2B_RigB_20171004T113838.bmp
12:    /mnt/spicy_4/daphnia/data/full_110647_D8_327_juju_2B_RigB_20171006T113335.bmp
13:    /mnt/spicy_4/daphnia/data/full_110675_D8_399_ctrl_2B_RigA_20171004T150839.bmp
14: /mnt/spicy_4/daphnia/data/full_110772_DBunk_321_ctrl_1A_RigB_20171004T170915.bmp
15: /mnt/spicy_4/daphnia/data/full_110779_DBunk_367_ctrl_2B_RigB_20171004T170413.bmp
16:    /mnt/spicy_4/daphnia/data/full_110851_D8_183_ctrl_1A_RigB_20171013T104805.bmp
17:  /mnt/spicy_4/daphnia/data/full_111177_DBunk_36_ctrl_1A_RigB_20171117T113603.bmp
18: /mnt/spicy_4/daphnia/data/full_111563_DBunk_298_juju_2A_RigB_20171130T101148.bmp
19: /mnt/spicy_4/daphnia/data/full_111564_DBunk_298_juju_2B_RigB_20171130T101642.bmp
20: /mnt/spicy_4/daphnia/data/full_111601_DBunk_590_juju_1A_RigB_20171130T112735.bmp
21: /mnt/spicy_4/daphnia/data/full_111626_DBunk_575_juju_1B_RigB_20171130T115941.bmp
22: /mnt/spicy_4/daphnia/data/full_111670_DBunk_295_ctrl_1B_RigA_20171130T094621.bmp
23: /mnt/spicy_4/daphnia/data/full_112033_DBunk_514_ctrl_1A_RigB_20171212T120443.bmp
24: /mnt/spicy_4/daphnia/data/full_112057_DBunk_561_ctrl_1A_RigB_20171212T122225.bmp
25: /mnt/spicy_4/daphnia/data/full_112058_DBunk_561_ctrl_1B_RigB_20171214T120324.bmp
26: /mnt/spicy_4/daphnia/data/full_112080_DBunk_579_juju_2B_RigB_20171212T161219.bmp

**Comment by awedwards** [comment]
Robert: Can you double check that the blur value is not saving? It is the "dorsal_edge_blur" value in the analysis_metadata file. It's saving fine for me, but there are a lot of values so I thought maybe you were looking at the wrong one?

I also am not having an issue with the masked regions not being saved. They're saving fine for me...

Dörthe:
It might be easier to explain over Skype, but I'll try to give an explanation here. Because of the size variation of the animals in the image, there is a fairly large range of the number of pixels that comprise the dorsal traces. If you recall, we do some interpolation for the traces with fewer pixels to reach 700 points total. The pictures you have listed likely naturally surpass the 700 pixels without any interpolation (the daphnia are probably fairly large in the images). So we can either pick a bigger number than 700 and do more interpolation on all of the images, or just deal with these few a different way.

**Comment by beckerdoerthe** [comment]
I think we can totally leave those <30 pics as they are, and then we might ignore them for the final procrustes analysis (or not). Robert just came across one of these (comment above), and I simply wanted to follow up on this, and whether those pics might make the GUI crash.

**Comment by awedwards** [comment]
Yeah that makes sense. I was thinking about it and I don't think it should matter how many points are in an individual dorsal edge. I don't have anything hardcoded in the GUI to only be handle 700 points, for instance.

**Comment by rjp5nc** [comment]
The image that was giving the error when saving was also within this list, by the way
